src/Common/isLocalAddress.cpp

```diff
@@ -91,16 +91,19 @@ struct NetworkInterfaces : public boost::noncopyable
     {
         static constexpr int NET_INTERFACE_VALID_PERIOD_MS = 30000;
         static NetworkInterfaces nf;
-        static auto last_updated_time = std::chrono::steady_clock::now();
+        static std::atomic<std::chrono::steady_clock::time_point> last_updated_time = std::chrono::steady_clock::now();
         static std::shared_mutex nf_mtx;
 
         auto now = std::chrono::steady_clock::now();
+        auto last_updated_time_snapshot = last_updated_time.load();
 
-        if (std::chrono::duration_cast<std::chrono::milliseconds>(now - last_updated_time).count() > NET_INTERFACE_VALID_PERIOD_MS)
+        if (std::chrono::duration_cast<std::chrono::milliseconds>(now - last_updated_time_snapshot).count() > NET_INTERFACE_VALID_PERIOD_MS)
         {
             std::unique_lock lock(nf_mtx);
+            if (last_updated_time.load() != last_updated_time_snapshot) /// it's possible that last_updated_time after we get the snapshot
```

Comment:
Actually, it's better to check here that the previous condition (now - last_update > valid_period) still holds. I'll merge it now and create a follow-up PR